Examples/TodoApp/TodoApp/src/tasks/TasksListViewModel.swift

```diff
@@ -0,0 +1,129 @@
+import Foundation
+import MobiusCore
+
+// Input from View
+protocol TasksListViewEventHandling {
+    func viewDidLoad()
+    func viewWillAppear()
+    func viewWillDisappear()
+    func didPressAddTaskButton()
+    func didCompleteTaskCreation(title: String, description: String)
+    var view: TaskViewing? { get set }
+}
+
+// Input from Effect Handler
+protocol TasksListViewModeling: AnyObject {
+    func loadTasks()
+    func save(task: Task)
+    func showAddTaskModal()
+}
+
+final class TasksListViewModel: TasksListViewEventHandling {
```

Comment:
I don't think `TasksListViewEventHandling`, `TasksListViewModeling`, or `TasksListViewModel` are bringing much value here.
The ViewController can handle starting and stopping the loop, the effect handler can perform side-effects directly, and the view already has an interface to the loop: TaskList.Event.

Reply:
I’d add `TaskViewing` to that list.

Reply:
Also, naming all protocols as gerunds isn’t idiomatic Swift.

Reply:
do you have a good name suggestion for `TaskViewing`? When I don't have a good one, I follow Apple guidelines